testdata/default/cheats/ArbitraryCheatcodes.t.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.18;
+
+import "ds-test/test.sol";
+import "cheats/Vm.sol";
+
+contract ArbitraryCheatcodesTest is DSTest {
+    Vm vm = Vm(HEVM_ADDRESS);
+
+    int128 constant min = -170141183460469231731687303715884105728;
+    int128 constant max = 170141183460469231731687303715884105727;
+
+    function test_int128() public {
+        int256 val = vm.arbitraryInt(16);
+        assertGe(val, min);
+        assertLe(val, max);
+    }
+
+    function testFail_int128() public {
+        int256 val = vm.arbitraryInt(16);
+        assertGt(val, max);
+    }
+
+    function test_address() public {
+        address fresh_address = vm.arbitraryAddress();
+        assert(fresh_address != address(this));
+        assert(fresh_address != address(vm));
+    }
+
+    function test_arbitraryUints(uint8 x) public {
+        vm.assume(0 < x);
+        vm.assume(x <= 32);
+        uint256 freshUint = vm.arbitraryUint(x);
```

Comment:
There seems to be a discrepancy between `test_arbitraryUints` and `test_arbitrarySymbolicWord` in terms of what the input to `vm.arbitraryUint()` represents. In `test_arbitraryUints`, it represents bytes (not bits), so the test asserts that `freshUint == vm.arbitraryUint(32) && freshUint <= type(uint256).max`. To make it consistent with `test_arbitrarySymbolicWord`, I think you'd need to change
```solidity
vm.assume(x <= 32);
...
 if (x == 32) {
...
  else {
    assert(freshUint <= 2 ** (8 * x) - 1);
...
```
to
```solidity
vm.assume(x <= 256);
...
 if (x == 256) {
...
  else {
    assert(freshUint <= 2 ** x - 1);
...
```

In Kontrol, we use bytes (e.g., `uint256 x = kevm.freshUint(32)`) and provide wrappers such as [freshUint256](https://github.com/runtimeverification/kontrol-cheatcodes/blob/ff5650a74fedada15c0bf9dd7300a9db3f432e4b/src/KontrolCheats.sol#L16) for uint and int denominations. But we can also adjust so it accepts bits instead, I agree that it's more intuitive.

Reply:
Similarly, in `test_int128` and `testFail_int128`, `vm.arbitraryInt(16)` is seemingly used to generate a value fitting into `int128`.

Reply:
thank you! changes in https://github.com/foundry-rs/foundry/pull/8882/commits/5220f88f8f728a627c5f1788444374625c17411d
- for `test_arbitraryUints` used `x = vm.arbitraryUint(0, 256);` to bound the value of bits to use (and avoid discarding fuzz runs using assume) and made the proper check
- for `test_int128` and `testFail_int128` used `int256 val = vm.arbitraryInt(128)` as suggested
- `length_limit` was not even used, so I removed it